<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>861</width>
    <height>466</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QGraphicsView" name="lSystemGraphicsView"/>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="lSystemEditor">
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllDockWidgetAreas</set>
   </property>
   <property name="windowTitle">
    <string>L System Editor</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QTabWidget" name="varAxiomTabWidget">
       <property name="tabShape">
        <enum>QTabWidget::Rounded</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tabVariable">
        <attribute name="title">
         <string>Variables</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QListView" name="variableListView">
           <property name="showDropIndicator" stdset="0">
            <bool>true</bool>
           </property>
           <property name="dragEnabled">
            <bool>true</bool>
           </property>
           <property name="dragDropOverwriteMode">
            <bool>true</bool>
           </property>
           <property name="dragDropMode">
            <enum>QAbstractItemView::InternalMove</enum>
           </property>
           <property name="defaultDropAction">
            <enum>Qt::MoveAction</enum>
           </property>
           <property name="alternatingRowColors">
            <bool>false</bool>
           </property>
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectItems</enum>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="variableButtonLayout">
           <item>
            <widget class="QPushButton" name="btnAddVariable">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnEditVariable">
             <property name="text">
              <string>Edit</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnRemoveVariable">
             <property name="text">
              <string>Remove</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="tabAxiom">
        <attribute name="title">
         <string>Axioms</string>
        </attribute>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QListView" name="axiomListView">
           <property name="enabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="axiomButtonLayout">
           <item>
            <widget class="QPushButton" name="btnAddAxiom">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnEditAxiom">
             <property name="text">
              <string>Edit</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btnRemoveAxiom">
             <property name="text">
              <string>Remove</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="rawOutput">
   <property name="windowTitle">
    <string>Output</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QListView" name="simulatedListView">
       <property name="flow">
        <enum>QListView::TopToBottom</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="simulationOptions">
   <property name="windowTitle">
    <string>Simulation Options</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="lblRecurseLevel">
         <property name="text">
          <string>Recurse Level</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbRecurseLevel">
         <property name="minimum">
          <number>1</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="btnSimulate">
       <property name="text">
        <string>Simulate</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>Generate Seed</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="lblSeed">
         <property name="text">
          <string>Seed</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="sbSeed">
         <property name="maximum">
          <number>999999999</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>861</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuWindow">
    <property name="title">
     <string>Window</string>
    </property>
    <widget class="QMenu" name="menuViews">
     <property name="title">
      <string>Views</string>
     </property>
     <addaction name="actionLSystem_Editor"/>
     <addaction name="actionOutput"/>
     <addaction name="actionSimulation_Options"/>
    </widget>
    <addaction name="menuViews"/>
   </widget>
   <addaction name="menuWindow"/>
  </widget>
  <action name="actionLSystem_Editor">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>LSystem Editor</string>
   </property>
  </action>
  <action name="actionOutput">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Output</string>
   </property>
  </action>
  <action name="actionSimulation_Options">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Simulation Options</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>lSystemEditor</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionLSystem_Editor</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLSystem_Editor</sender>
   <signal>toggled(bool)</signal>
   <receiver>lSystemEditor</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>148</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rawOutput</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionOutput</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOutput</sender>
   <signal>toggled(bool)</signal>
   <receiver>rawOutput</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>334</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSimulation_Options</sender>
   <signal>toggled(bool)</signal>
   <receiver>simulationOptions</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>64</x>
     <y>371</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulationOptions</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>actionSimulation_Options</receiver>
   <slot>setChecked(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>64</x>
     <y>371</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
